lib/export/src/Export.hs

```diff
@@ -1271,3 +1274,33 @@ makeAnnotations thy p = res
       if isNothing (List.find (== "locations-report") (theoryBuiltins thy))
         then pr
         else translateTermsReport pr
+
+-- | Pull out nots in formula
+pullnots :: LNFormula -> Either LNFormula LNFormula
+pullnots fm = 
+  let fm_partially_rewritten = pullnots' fm in
+  if notsArePulledOut fm_partially_rewritten 
+    then Right fm_partially_rewritten -- in this case, formula is fully rewritten
+    else Left fm_partially_rewritten  -- Error with partially rewritten formula
+  where
+    notsArePulledOut (Not p) = notsArePulledOut' p -- top-level not expected
+    notsArePulledOut _       =  False
+
+    notsArePulledOut' (Not _)      = False -- check that there are  no nots below top level
+    notsArePulledOut' (Conn _ p q) = notsArePulledOut' p && notsArePulledOut' q
+    notsArePulledOut' (Qua _ _ p ) = notsArePulledOut' p
+    notsArePulledOut' _            = True
+
+    pullnots' f = if f /= pullStep f then pullnots' (pullStep f) else f
+
+    pullStep fm' = case fm' of
+      Conn And (Not p) (Not q)  -> Not $ p .||. q
+      Conn Or (Not p) (Not q)   -> Not $ p .&&. q
+      Conn Imp p (Not q)        -> Not $ p .&&. q
+      Conn c p q                -> Conn c (pullStep p) (pullStep q)
```

Comment:
We should exhaustively list the cases here, so if new connectives are added, we do not compute incorrect results. I think this can only be an `iff`, in which case ... can't we pull the Not out as well, if both sides have it?

Reply:
Yes, the `Iff` case should absolutely be there - I think that I missed it only because I somehow didn't write test formulas with it.

Reply:
Yup that's good.